apps/prairielearn/src/lib/assessment.sql

```diff
@@ -50,6 +50,51 @@ WHERE
   a.id = $assessment_id
   AND ai.open;
 
+-- BLOCK close_assessment_instance
+WITH
+  updated_assessment_instance AS (
+    UPDATE assessment_instances AS ai
+    SET
+      open = FALSE,
+      closed_at = CURRENT_TIMESTAMP,
+      duration = assessment_instances_duration (ai.id),
+      modified_at = now(),
+      -- Mark the assessment instance as in need of grading. We'll start
+      -- grading immediately, but in case the PrairieLearn process dies in
+      -- the middle of grading, the `autoFinishExams` cronjob will grade the
+      -- assessment instance at some point in the near future.
+      grading_needed = TRUE
+    WHERE
+      ai.id = $assessment_instance_id
+    RETURNING
+      ai.id
+  )
+INSERT INTO
+  assessment_state_logs (open, assessment_instance_id, auth_user_id)
+SELECT
+  FALSE,
+  updated_assessment_instance.id,
+  $authn_user_id
+FROM
+  updated_assessment_instance;
+
+-- BLOCK select_variants_for_assessment_instance_grading
+SELECT DISTINCT
+  ON (iq.id) to_jsonb(v.*) AS variant,
```

Comment:
This formatting of `DISTINCT ON` bothers me a bit, but it's a problem with sql-formatter... https://github.com/sql-formatter-org/sql-formatter/issues/144